src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -100,42 +101,65 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
                                                   max_context_len->output(0));
     manager.register_pass<PrevSequenceLengthPattern>(prev_max_seq_len, batch_dim);
     manager.register_pass<TotalSequenceLengthPattern>(max_context_len);
-
     manager.register_pass<PositionIDsReplacer>(unsqueezed_position_ids->output(0));
-
     manager.run_passes(model);
 
+    {
+        // Remove all Assigns aggressively, the path from the kv-cache concat to Assign can be complicated,
+        // but there is no reason to track it and reject part of the Assigns, because the model will remain
+        // in incorrect form anyway.
+        auto sinks = model->get_sinks();
+
+        for (auto& sink : sinks) {
+            model->remove_sink(sink);
+        }
+    }
+
+    for (auto& result : results_to_remove) {
+        model->remove_result(result);
+    }
+
     if (has_parameter(model, "beam_idx")) {
-        if (const auto& parameter =
+        if (const auto& beam_idx =
                 std::dynamic_pointer_cast<v0::Parameter>(model->input("beam_idx").get_node_shared_ptr())) {
-            model->remove_parameter(parameter);
+            model->remove_parameter(beam_idx);
+
+            std::stringstream consumers;
+            consumers << std::endl;
+            for (auto& input : beam_idx->output(0).get_target_inputs()) {
+                consumers << *input.get_node() << std::endl;
+            }
+            OPENVINO_ASSERT(beam_idx->output(0).get_target_inputs().size() == 0,
+                            "PagedAttention transformation failed: couldn't remove ",
+                            beam_idx->output(0).get_target_inputs().size(),
+                            " inputs of 'beam_idx' input: ",
+                            consumers.str());
         } else {
             return false;
         }
     }
 
-    if (const auto& parameter =
+    if (const auto& attn_mask =
             std::dynamic_pointer_cast<v0::Parameter>(model->input("attention_mask").get_node_shared_ptr())) {
-        model->remove_parameter(parameter);
+        model->remove_parameter(attn_mask);
+
+        std::stringstream consumers;
```

Comment:
is's always created, but will be used only if `attn_mask->output(0).get_target_inputs().size() != 0`
it's better to create this variable inside `if { ... }`